OrbitQt/orbitsamplingreport.cpp

```diff
@@ -86,6 +89,21 @@ void OrbitSamplingReport::on_PreviousCallstackButton_clicked() {
   RefreshCallstackView();
 }
 
+void OrbitSamplingReport::OnCurrentThreadTabChanged(int current_tab_index) {
+  if (current_tab_index == -1) {
+    return;
+  }
+
+  auto treeView = m_OrbitDataViews.at(current_tab_index);
+  QModelIndexList index_list = treeView->GetTreeView()->selectionModel()->selectedIndexes();
+  if (!index_list.isEmpty() && index_list.front().isValid()) {
+    treeView->GetTreeView()->GetModel()->OnRowSelected(index_list.front().row());
+  } else {
+    m_SamplingReport->ClearReport();
```

Comment:
Is this correct? Shouldn't it just clear the callstack selection, not the complete sampling report?

Reply:
Thanks for raising this problem. The name might be misleading. `m_SamplingReport->ClearReport()` only clears the callstack report rather than the whole sampling report.